homeassistant/components/zha/config_flow.py

```diff
@@ -91,6 +93,31 @@ async def async_step_pick_radio(self, user_input=None):
             data_schema=vol.Schema(schema),
         )
 
+    async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
+        """Handle zeroconf discovery."""
+        # Hostname is format: livingroom.local.
+        local_name = discovery_info["hostname"][:-1]
+        node_name = local_name[: -len(".local")]
+
+        # Check if already configured
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        await self.async_set_unique_id(node_name)
```

Comment:
Does it make sense to connect to the `ip` address here instead like we do with esphome to avoid resolution / mdns lookup delays at startup time.

```suggestion
        await self.async_set_unique_id(node_name, updates={CONF_DEVICE: ... new ip...})
```

Reply:
Unless IP address would change. There were at least two different people asking how to change the IP address for the serial-over-tcp config entry.
Or we could store both and try to establish connection directly to IP and then falling back to mdns resolution??? not sure if the additional complexity worth it 🤷

Reply:
`esphome` used to have issues about not being able to connect to the device on startup until we switched the default behavior to create the entry with the ip, and then update it (https://github.com/home-assistant/core/blob/dev/homeassistant/components/esphome/config_flow.py#L95) when we get a zeroconf discovery.